### PR TITLE
fix(chat): reset transcript view on session switch

### LINKED ISSUE #936: bug(chat): Cmd+T new chat tab shows a stale transcript from the previously-viewed session
## Summary

`Cmd/Ctrl+T` (`global.new-tab`) creates a fresh **"New chat"** tab, but the chat panel renders a **stale transcript from a previously-viewed session** inside it. Observed: a brand-new "New chat" tab in the `feat-dealflow-approve-records` workspace (nyc-real-estate) displayed a full conversation that belonged to a **different workspace's** session (`fix-915-devshell-path`, the env-nix-devshell debugging transcript, `16 in · 13.5k out`).

## Key observation — this is a stale-render bug, not DB content

The new tab shows **both** the old transcript **and** the empty-state placeholder ("`>`  Send a message to start a conversation") at the same time, with the composer reading "Send a message…".

That co-presence is diagnostic: the empty-state only renders when the active session genuinely has **zero messages**. So the new session *is* empty in store state — `chatMessages[newSessionId]` is `[]` — yet the transcript DOM from the previously-viewed session is **still on screen**. This is a **frontend stale-render / not-reset-on-session-switch bug**, not chat_messages being wrongly populated for the new session.

(It is therefore *not* the multi-instance `chat_messages` cross-pollution flagged in #935 — if the DB had returned rows for the new session, the empty-state would not show. The multi-instance setup is still worth ruling out as a confounder, but the symptom here points squarely at the chat-panel render path.)

## How `Cmd+T` works (researched)

`global.new-tab` → `mod+t` → `executeNewTab` (`src/ui/src/hotkeys/contextActions.ts:168`):

```ts
const session = await deps.createChatSession(wsId);
const post = useAppStore.getState();
if (post.selectedWorkspaceId !== wsId) return;   // workspace switched → drop
post.addChatSession(session);
post.selectSession(wsId, session.id);
```

A fresh session id is created, added, and selected. `ChatPanel.tsx` then derives:

```ts
const activeSessionId = s.selectedSessionIdByWorkspaceId[s.selectedWorkspaceId] ?? null;
const messages = activeSessionId ? chatMessages[activeSessionId] || [] : [];
```

For the new session `chatMessages[newId]` is `undefined` → `messages` is `[]`. So `ChatPanel`'s `messages` is correctly empty — which means the visible transcript is **not** coming from `messages`. The stale transcript is rendered by something that did **not** reset when `activeSessionId` changed.

## Likely root cause (to confirm)

Candidates, in order of suspicion:

1. **The message-list / transcript view holds the previous render and is not keyed on `activeSessionId`.** If the transcript subtree (or a virtualized list) isn't `key={activeSessionId}`-reset or doesn't clear on session change, its prior DOM/state survives the switch. The fix: key the transcript region on the session id (or explicitly clear) so switching sessions always remounts an empty list for an empty session.
2. **A late, mis-targeted history load.** If switching to the new session kicks a history fetch and a *previous* session's in-flight history load resolves afterwards and writes under the wrong id (no session-token guard), the old transcript leaks in. (SessionTabs already has an in-flight-invalidation note ~line 207 — verify the same guard exists on the panel's history-load path.)
3. **Transcript sourced from a non-`messages` store slice keyed wrong** — e.g. `completedTurns`, `streamingContent`, or a memo in a child component that didn't invalidate on `activeSessionId`.

Recent chat-panel changes touching session/workspace switching (e.g. #913 "preserve steering prompt preview during workspace switches") are worth checking as a regression point.

## Reproduction

1. Open a workspace, view a chat session with a transcript.
2. Switch to a different workspace.
3. `Cmd+T` to open a "New chat" tab.
4. Observe: the new tab shows the previous session's transcript above the "start a conversation" empty-state.

## Acceptance criteria

- [ ] A `Cmd+T` "New chat" tab renders an empty transcript — no content from any prior session/workspace.
- [ ] Switching between sessions/tabs never leaves a previous session's transcript visible under the new session.
- [ ] Any in-flight chat-history load is guarded by a session token so a late resolve cannot write to the wrong session.
- [ ] Test: select a session with messages → create a new session → assert the rendered transcript for the new session is empty.

## Notes for triage

- Frontend-only: `ChatPanel.tsx` (god file — fix by keying/extracting the transcript region, not by adding inline branches) and possibly the chat-history load path. No backend/DB change expected.
- The maintainer is currently running multiple Claudette instances on one shared DB (see #935). That is a separate hazard; this symptom is explained by the stale-render path above and should reproduce with a single instance.

### PR DESCRIPTION
**Pull request by jamesbrink**
## Summary

- Reset the chat transcript renderer when the active chat session changes so stale message DOM/state cannot survive a tab switch.
- Guard chat history, completed-turn, and checkpoint hydration callbacks so late async responses only apply while their workspace/session is still selected.
- Add a regression test that simulates a sticky transcript renderer and verifies a newly selected empty chat stays clear of the previous transcript.

## Root Cause

Issue #936 showed a new `Cmd/Ctrl+T` chat tab rendering both the empty-state placeholder and a previous session's transcript. The store path already resolved the new session to an empty message list, so the bug was in the frontend render/hydration path rather than persisted chat rows.

The fix keys the session-specific transcript surfaces by `activeSessionId` and adds a live selection token check around async hydration callbacks. That covers both the stale child-renderer case and the race where an older history request resolves after the user has moved to a different session.

## Validation

- `cd src/ui && bun run test ChatPanel.staleTranscript.test.tsx`
- `cd src/ui && bunx tsc -b`
- `cd src/ui && bun run lint` (passes with existing warnings only)
- `cd src/ui && bun run test`
- `cd src/ui && bun run build`

Closes #936

### CODE REVIEW

File: src/ui/src/components/chat/ChatPanel.staleTranscript.test.tsx [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+vi.mock("react-i18next", () => ({
+  useTranslation: () => ({
+    t: (key: string, fallbackOrOptions?: string | Record<string, unknown>) =>
+      typeof fallbackOrOptions === "string" ? fallbackOrOptions : key,
+  }),
+}));
+
+vi.mock("../../services/tauri", () => serviceMocks);
+
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/ChatPanel.staleTranscript.test.tsx]: Fixed in 68049908. The test now uses a partial `services/tauri` mock that spreads the real barrel exports and only overrides the Tauri calls this harness needs, so ChatPanel helpers such as `chatMessageDispatch` and `applySelectedModel` keep their expected imports. Re-ran `cd src/ui && bun run test ChatPanel.staleTranscript.test.tsx` and `cd src/ui && bunx tsc -b`.
